[arb] Arb package cannot be used to build programs

**Describe the bug**
The `arb` package cannot be used to build programs because linking fails.

**Environment**
- OS: Windows 10
- Compiler: Visual Studio 17.0.1 2022

**To Reproduce**

`main.cpp`:

```cpp
#include <malloc.h>
#include <arb.h>

int main()
{
    flint_printf("Computed with arb-%s\n", arb_version);
}
```

`vcpkg.json`:

```json
{
  "name": "testarb",
  "version-string": "0.1.0",
  "dependencies": [
    "arb"
  ]
}
```

`CMakeLists.txt`:

```cmake
cmake_minimum_required(VERSION 3.15)

project(testarb CXX)

include_directories("./vcpkg/packages/arb_x64-windows/include")
include_directories("./vcpkg/packages/flint_x64-windows/include")
include_directories("./vcpkg/packages/gmp_x64-windows/include")
include_directories("./vcpkg/packages/mpfr_x64-windows/include")
include_directories("./vcpkg/packages/pthreads_x64-windows/include")

link_directories("./vcpkg/packages/arb_x64-windows/lib")
link_directories("./vcpkg/packages/flint_x64-windows/lib")
link_directories("./vcpkg/packages/gmp_x64-windows/lib")
link_directories("./vcpkg/packages/mpfr_x64-windows/lib")
link_directories("./vcpkg/packages/pthreads_x64-windows/lib")

add_executable(testarb main.cpp)
target_compile_features(testarb PRIVATE cxx_std_20)

target_link_libraries(testarb PRIVATE gmp flint mpfr arb)
```

Command line:

```
git clone https://github.com/Microsoft/vcpkg.git
.\vcpkg\bootstrap-vcpkg.bat
cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=./vcpkg/scripts/buildsystems/vcpkg.cmake
cmake --build build --config Release
```

Output:

```
main.obj : error LNK2019: unresolved external symbol __imp_arb_version referenced in function main [C:\work\testarb\bui
ld\testarb.vcxproj]
```

**Expected behavior**
Should link.

Thanks. Here is perhaps a better CMake file. 

```cmake
cmake_minimum_required(VERSION 3.15)

project(testarb CXX)

find_path(ARB_INCLUDE_DIR arb.h)
find_library(ARB_LIBRARY NAMES arb)

find_path(FLINT_INCLUDE_DIR flint/flint.h)
find_library(FLINT_LIBRARY NAMES flint)

find_path(GMP_INCLUDE_DIR gmp.h)
find_library(GMP_LIBRARY NAMES gmp)

find_path(MPFR_INCLUDE_DIR mpfr.h)
find_library(MPFR_LIBRARY NAMES mpfr)

find_path(PTHREADS_INCLUDE_DIR pthread.h)
find_library(PTHREADS_LIBRARY NAMES pthreadVC3)

add_executable(testarb main.cpp)
target_compile_features(testarb PRIVATE cxx_std_20)

target_include_directories(testarb PRIVATE 
    "${ARB_INCLUDE_DIR}" 
    "${FLINT_INCLUDE_DIR}"
    "${GMP_INCLUDE_DIR}"
    "${MPFR_INCLUDE_DIR}"
    "${PTHREADS_INCLUDE_DIR}"
)

target_link_libraries(testarb PRIVATE 
    "${ARB_LIBRARY}"
    "${FLINT_LIBRARY}"
    "${GMP_LIBRARY}"
    "${MPFR_LIBRARY}"
    "${PTHREADS_LIBRARY}"
)

message(STATUS "Include directories:")
message(STATUS "${ARB_INCLUDE_DIR}")
message(STATUS "${FLINT_INCLUDE_DIR}")
message(STATUS "${GMP_INCLUDE_DIR}")
message(STATUS "${MPFR_INCLUDE_DIR}")
message(STATUS "${PTHREADS_INCLUDE_DIR}")

message(STATUS "Libraries:")
message(STATUS "${ARB_LIBRARY}")
message(STATUS "${FLINT_LIBRARY}")
message(STATUS "${GMP_LIBRARY}")
message(STATUS "${MPFR_LIBRARY}")
message(STATUS "${PTHREADS_LIBRARY}")
```

The original issue still remains: `arb` won't link.

```
error LNK2019: unresolved external symbol __imp_arb_version referenced in function main [C:\code\testarb\bui
ld\testarb.vcxproj]
```

> The original issue still remains: `arb` won't link.
> 
> ```
> error LNK2019: unresolved external symbol __imp_arb_version referenced in function main [C:\code\testarb\bui
> ld\testarb.vcxproj]
> ```

Hm, port `arb` supports only static linkage, regardless of triplet. IIUC there shouldn't be an `__imp_` prefix on symbol names from the `arb` headers. There seem to be a `declspec` issue in the installed headers.

--- a/ports/arb/portfile.cmake
+++ b/ports/arb/portfile.cmake
@@ -1,5 +1,3 @@
-vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO fredrik-johansson/arb

--- a/ports/arb/vcpkg.json
+++ b/ports/arb/vcpkg.json
@@ -1,8 +1,10 @@
 {
   "name": "arb",
   "version": "2.21.1",
+  "port-version": 1,
   "description": "a C library for arbitrary-precision interval arithmetic",
   "homepage": "https://github.com/fredrik-johansson/arb",
+  "license": "LGPL-2.1",
   "dependencies": [
     "flint",
     {

--- a/versions/a-/arb.json
+++ b/versions/a-/arb.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "e1b560a5a2aa835d54da9fdabac5eea48255e93d",
+      "version": "2.21.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "06326497117fb757651649225284d2fe4100ef79",
       "version": "2.21.1",
       "port-version": 0

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -122,7 +122,7 @@
     },
     "arb": {
       "baseline": "2.21.1",
-      "port-version": 0
+      "port-version": 1
     },
     "arcus": {
       "baseline": "4.10.0",

